stdlib/seq.mc

```diff
@@ -14,14 +14,26 @@ let foldl = fix (lam foldl. lam f. lam acc. lam seq.
 )
 let foldl1 = lam f. lam l. foldl f (head l) (tail l)
 
+let foldr = fix (lam foldr. lam f. lam acc. lam seq.
+    if null seq
+    then acc
+    else f (foldr f acc (tail seq)) (head seq)
```

Comment:
Should the arguments to `f` be flipped? My feeling is that `foldr cons [] [1,2,3]` should be `[1,2,3]`, whereas now it is an error.

Reply:
I guess we could, but I think the above definition is consistent with foldl, i.e. `f : acc -> x -> acc` so `foldl cons [] [1,2,3]` also gives an error.

Reply:
No your right, for `foldr` `f: x -> acc -> acc` but for `foldl` `f: acc -> x -> acc`.